# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1351329631 -7200
#      Sat Oct 27 11:20:31 2012 +0200
# Node ID 81ff500bfb4e7e3f0fa3b7e66948e520e95dc387
# Parent  ed6385e234208426909ea67fa5e78179d62e73e2
fix indication of terminal widget visibility in window menu

* terminal_dock_widget.cc: added handler closeEvent (),
    moved implementation of handle_visibility_changed () here
* terminal_dock_widget.h: added handler closeEvent (),
    moved implemenatiotn of handle_visibility_changed () to .cc

diff --git a/libgui/src/terminal-dockwidget.cc b/libgui/src/terminal-dockwidget.cc
--- a/libgui/src/terminal-dockwidget.cc
+++ b/libgui/src/terminal-dockwidget.cc
@@ -34,16 +34,30 @@ terminal_dock_widget::terminal_dock_widg
   setWindowTitle (tr ("Command Window"));
   setWidget (terminal);
 
   connect (this, SIGNAL (visibilityChanged (bool)), this, SLOT (handle_visibility_changed (bool)));
   // topLevelChanged is emitted when floating property changes (floating = true)
   connect (this, SIGNAL (topLevelChanged(bool)), this, SLOT(top_level_changed(bool)));
 }
 
+void
+terminal_dock_widget::closeEvent (QCloseEvent *e)
+{
+  emit active_changed (false);
+  QDockWidget::closeEvent (e);
+}
+
+void
+terminal_dock_widget::handle_visibility_changed (bool visible)
+{
+  if (visible)
+    emit active_changed (true);
+}
+
 // slot for signal that is emitted when floating property changes
 void
 terminal_dock_widget::top_level_changed (bool floating)
 {
   if(floating)
     {
       setWindowFlags(Qt::Window);  // make a window from the widget when floating
       show();                      // make it visible again since setWindowFlags hides it
diff --git a/libgui/src/terminal-dockwidget.h b/libgui/src/terminal-dockwidget.h
--- a/libgui/src/terminal-dockwidget.h
+++ b/libgui/src/terminal-dockwidget.h
@@ -31,18 +31,18 @@ class terminal_dock_widget : public QDoc
   Q_OBJECT
   public:
   terminal_dock_widget (QTerminal *terminal, QWidget *parent = 0);
 
 signals:
   void active_changed (bool active);
 
 public slots:
-  void handle_visibility_changed (bool visible)
-  {
-    if (visible)
-      emit active_changed (true);
-  }
+  void handle_visibility_changed (bool visible);
   /** Slot when floating property changes */
   void top_level_changed (bool floating);
+
+protected:
+  void closeEvent (QCloseEvent *event);
 };
 
+
 #endif // TERMINALDOCKWIDGET_H
